mono-traversable/test/Main.hs

```diff
@@ -149,6 +149,26 @@ main = hspec $ do
         prop "works on lists" $ \(Positive i) j ->
             ocompareLength (replicate i () :: [()]) j @?= compare i j
 
+    describe "groupBy" $ do
+        let test name dummy = prop name $ \(QCM.NonEmpty xs) -> do
+                let seq' = fromListAs xs dummy
+                let listDef f = Prelude.fmap fromList . List.groupBy f . otoList
+                groupBy (==) seq' @?= listDef (==) seq'
+                groupBy (/=) seq' @?= listDef (/=) seq'
+                groupBy (<) seq' @?= listDef (<) seq'
+                groupBy (>) seq' @?= listDef (>) seq'
```

Comment:
What do you think about also adding `groupBy g seq' @?= listDef g seq'` where `g` is generated by QC like `prop name $ \xs (Fn2 g) -> do`?

Reply:
That is fine. I was not aware that there was an ability to automatically randomly generate functions. I would still like to keep the other functions as these are functions that would commonly be used.